app/models/api_token.rb

```diff
@@ -0,0 +1,27 @@
+class APIToken < ApplicationRecord
+  has_secure_token :hashed_token, length: 32
+  encrypts :hashed_token, deterministic: true
+
+  belongs_to :lead_provider, optional: true
+
+  enum :scope, {
```

Comment:
do we have other options of this implementation, as this is similar to NPQ?

Reply:
Hi @cwrw I've added a new option using polymorphic association